3_javascript/javascript1/week2/js_w2.js

```diff
@@ -0,0 +1,122 @@
+////////////////////////Flight booking fullname function////////////////////////
+
+/////// step1
+// function getFullname(firstname, surname) {
+//     let fullname = firstname + surname;
+//     return fullname;
+//   }
+  
+//   let fullname1 = getFullname("Meraj ", "Sharifi");
+//   let fullname2 = getFullname("Mina ", "Miri");
+  
+//   console.log("Full Name 1:", fullname1);
+//   console.log("Full Name 2:", fullname2);
+  
+
+/////// step2 - Formal fullname
+
+// function getFullname(firstname, surname, useFormalName) {
+//     if (useFormalName) {return "Lord " + firstname + surname;}
+//     else {return firstname + surname;}
+//   }
+  
+//   let formalFullName = getFullname("Meraj ", "Sharifi", true);
+//   let inFormalFullName = getFullname("Mina ", "Miri", false);
+  
+//   console.log("Formal Full Name:", formalFullName);
+//   console.log("Informal Full Name:", inFormalFullName);
+
+  
+/////// FINAL STEP: WITHOUT NAME AND ADDING FEMALE NAME
+
+  function getFullname(firstname, surname, useFormalName, isFemale) {
+    if (useFormalName) {return "Lord " + firstname + surname}
+    else if (!firstname && !surname) {return "Please provide a valid name.";}
+    else if (isFemale) {return "Lady " + firstname + surname}
+    else {return firstname + surname}
+}
+
+  const formalFullName = getFullname("Meraj ", "Sharifi", true, false);
+  const inFormalFullName = getFullname("Mina ", "Miri", false, false);
+  const formalFullFemaleName = getFullname("Sara ", "Ono", false, true);
+  const missingNames = getFullname("", "", false); // Empty names
+
+  console.log("Formal Full Name:", formalFullName);
+  console.log("Informal Full Name:", inFormalFullName);
+  console.log("Informal Full Female Name:", formalFullFemaleName);
+  console.log("Missing Names:", missingNames);
+
+  ////////////////////////////////////////////////////////////
+  ////////////////////////Event application////////////////////////
+
+function getEventWeekday (daysFromToday) {
+    const weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    const today = new Date();
+    const eventDayIndex = (today.getDay() + daysFromToday) % 7;
+    const positiveIndex = (eventDayIndex + 7) % 7;
+    const eventDayOfWeek = weekdays[positiveIndex];
+    return eventDayOfWeek;
+}
+
+const eventDayOfWeek = getEventWeekday (5);
+
+console.log(`The event will be on ${eventDayOfWeek}`)
+
```

Comment:
🎉 Great job taking on this coding challenge
💪 Everything looks clean and works like a charm
🚀 Awesome job with using `const`